app/services/staff_nomis_checker.rb

```diff
@@ -14,10 +18,10 @@ def prisoner_existance_status
     return NOT_LIVE unless prisoner_check_enabled?
 
     case prisoner_existance_error
-    when nil
-      VALID
-    when UNKNOWN
-      UNKNOWN
+    when nil              then VALID
```

Comment:
I find this confusing; it mixes the prisoner existence with location and depends on the first sorted error message.

The location is almost the same as the prisoner availability in terms of how it interacts with prisoner existence:

- If prisoner existence is invalid then location is unknown: The user could have typoed the noms_id but still be valid (and existence be invalid because the date of birth does not match). If we lookup the location then it will be the location of a completely different prisoner.

- If the prisoner existence is unknown then the location is unknown. Similar to above but the existence api times out, responds with a non 200 code.

- Prisoner existence is valid, then location could be valid, unknown or invalid